tests/test_unstable/conftest.py

```diff
@@ -202,3 +202,14 @@ def log_messages_task(ctx: TaskContext) -> None:
             ctx.warning("This is a warning message.")
 
         self.add_task(StartupTask(name="log_task", target=log_messages_task))
+
+
+@pytest.fixture(autouse=True)
+def reset_singleton() -> Generator[None, None, None]:
```

Comment:
It's possible to use the slightly shorter notation:
```suggestion
def reset_singleton() -> Iterator[None]:
```

I would also put this `autouse=True` fixture at the top of the file since it runs before and after every single test case and thus should be very visible 😄 

Last note, I would typically reach for `monkeypatch.setattr` to temporarily override attributes for tests (but no need to use here, as you also want to nuke it as part of test teardown).

Reply:
added this